src/trix/models/document.coffee

```diff
@@ -125,17 +125,34 @@ class Trix.Document extends Trix.Object
 
     text = leftText.appendText(rightText)
 
-    removingLeftBlock = leftIndex isnt rightIndex and leftLocation.offset is 0
-    useRightBlock = removingLeftBlock and leftBlock.getAttributeLevel() >= rightBlock.getAttributeLevel()
-
-    if useRightBlock
+    previousCharacterRange = [startPosition - 1, endPosition - 1]
+    previousCharacter = rightBlock.text.getTextAtRange(previousCharacterRange).toString()
+    currentCharacter = @getCharacterAtPosition(startPosition, endPosition)
+    nextCharacter = @getCharacterAtPosition(endPosition, endPosition + 1)
+
+    hasBlockAttributes = @getBlockAtIndex(rightIndex).getAttributes().length > 0
+    isEmptyBlock = @getBlockAtIndex(rightIndex).text.toString() is "\n"
+    multipleBlocksInRange = leftIndex isnt rightIndex
+
+    removingLeftBlock = multipleBlocksInRange and
+                        leftBlock.getAttributeLevel() >= rightBlock.getAttributeLevel() and
+                        leftLocation.offset is 0
+    removingNewline = currentCharacter is "\n" and nextCharacter is "\n" and not hasBlockAttributes and not isEmptyBlock
+
+    if removingNewline
```

Comment:
Can we catch this condition early on and just return a new document with the newline removed? I think it would be easier to follow that way instead of setting up the right blocks/indexes for splicing.